### PR TITLE
feat: infer release type from commits since last tag

### PR DESCRIPTION
**Pull request by ursinha**
I've done some pretty non-orthodox things but it solves the problem pretty ok. 

Introduced behaviors:
- This version only creates releases when there's a version bump
- Version bumps happen for feat, fix, chore, refactor; everything else is not worth a release
- Introduced ability to use 3rd party tool to generate fancy changelogs (hello changelogen)

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -20,27 +24,54 @@ async function run(): Promise<void> {
         if (currentVersion === null)
             return
 
-        const lastCommitMessage = await getLastCommitMessage()
-        if (lastCommitMessage === null)
-            return
-
-        const releaseType = getReleaseTypeFromCommitMessage(lastCommitMessage)
+        let releaseType: string | null = 'non-release'
```
Comment (Guiqft) [inline src/main.ts]: release type still can be null even with a default value?

Comment (Guiqft) [reply src/main.ts]: also, this could be 
```suggestion
        let releaseType: ReleaseType = 'non-release'
```

File: src/main.ts [hunk 2]
```diff
@@ -20,27 +24,54 @@ async function run(): Promise<void> {
...
 
-        // If the commit isn't of type `feat` or `fix`, we don't want to bump the version
-        if (releaseType !== null) {
+        // If we're creating a release we want to bump the version
+        if (releaseType !== null && releaseType !== 'non-release') {
+            core.info(`New release version should be type ${releaseType}`)
+            core.setOutput('release-type', releaseType)
             const nextVersion = isReleaseCandidate
-                ? getNextVersion({ currentVersion, releaseType })
+                ? getNextVersion({ currentVersion, releaseType: releaseType as ReleaseType })
```
Comment (Guiqft) [inline src/main.ts]: You won't need the type casting if you fix the type on let releaseType

File: src/main.ts (hunk 1 above)
Comment (ursinha) [reply src/main.ts]: After some quick deliberation, we agreed on:
```
 let releaseType: ReleaseType | null = null
```

File: src/main.ts (hunk 2 above)
Comment (ursinha) [reply src/main.ts]: Done

**Pull request merged**